codebase-pipelines/buildspec-deploy.yml

```diff
@@ -75,92 +84,101 @@ phases:
 
       # Get service name
       - service_name=$(aws ecs list-services --cluster "${cluster}" | jq -r '.serviceArns[] | select(test("'${cluster}'-'${SERVICE}'($|-Service.*)"))' | cut -d '/' -f3)
-
-      # Update task definition
-      - task_definition=$(aws ecs describe-task-definition --task-definition "${task_family}")
-      - new_task_definition=$(echo ${task_definition} | jq '.taskDefinition | .containerDefinitions |= map(if .name == '\"${SERVICE}\"' then .image = '\"${image_uri}\"' else . end) | del(.taskDefinitionArn) | del(.revision) | del(.status) | del(.requiresAttributes) | del(.compatibilities) |  del(.registeredAt)  | del(.registeredBy)')
-      - new_task_info=$(aws ecs register-task-definition --cli-input-json "${new_task_definition}")
-      - new_revision=$(echo ${new_task_info} | jq '.taskDefinition.revision')
-
-      # Get desired task count
-      - count=0
-      - cd "${CODEBUILD_SRC_DIR}"
-
-      # If count is a range, get the first value otherwise get count
+      
+      # Run Copilot package command and clean the manifest output
+      - copilot svc package --env "${ENVIRONMENT}" --name "${SERVICE}" | yq .Metadata.Manifest | jq -r . | yq 'del(.image.location)' > new.yml
+
+      # Get the current CloudFormation template and clean it for comparison
+      - aws cloudformation get-template --stack-name "${APPLICATION}-${ENVIRONMENT}-${SERVICE}" --output json --query TemplateBody | jq -r . | yq .Metadata.Manifest | jq -r . | yq 'del(.image.location)' > current.yml
```

Comment:
Also wondering about the mix of yq and jq in here. 

The output from the cloudformation command is specified as json, so don't think we need the yq in there at all.
Perhaps with just a final yq command to reformat it as standard YAML prior to the diff.

Reply:
This works for me locally:
`--output json --query TemplateBody | jq -r . | yq .Metadata.Manifest | yq 'del(.image.location)'` 

The initial "jq" is needed to get the unquoted YAML out of the JSON response. The second one isn't necessary as it's YAML at that point and so fails as jq can't parse it.

Reply:
To address both comments.

The included commands run on yq as installed from PyPi, currently the latest version on PyPi is 3.4.3
This version of yq allows use of jq and yq together to filter down to the required template data.

This has been tested locally with 3.4.3 & works as expected.

Later versions of yq do not support the use of yq and jq as is present in the buildspec commands. 
PyPi has been on 3.4.3 as its latest offering since April 2024, so unless we source a different version of yq or PyPi updates then we should continue using the buildspec commands with a mix of jq and yq